smarts/core/scenario.py

```diff
@@ -551,7 +551,7 @@ def _extract_mission(mission, road_map):
 
         def resolve_offset(offset, lane_length):
             # epsilon to ensure we are within this edge and not the subsequent one
-            epsilon = 1e-6
+            epsilon = 2 if lane_length > 2 else 1e-6
```

Comment:
I think there is a bigger issue here, the problem this tried to address is that injecting a vehicle so close to the beginning of the lane would potentially inject on top of another vehicle but this is not the way to fix that issue.

Reply:
Do you want to file an issue about this?

Reply:
https://github.com/huawei-noah/SMARTS/issues/1217